rleplus/internal/bitvector.go

```diff
@@ -0,0 +1,136 @@
+package bitvector
+
+import "errors"
+
+var (
+	// ErrOutOfRange - the index passed is out of range for the BitVector
+	ErrOutOfRange = errors.New("index out of range")
+)
+
+// BitNumbering indicates the ordering of bits, either
+// least-significant bit in position 0, or most-significant bit
+// in position 0.
+//
+// It it used in 3 ways with BitVector:
+// 1. Ordering of bits within the Buf []byte structure
+// 2. What order to add bits when using Extend()
+// 3. What order to read bits when using Take()
+//
+// https://en.wikipedia.org/wiki/Bit_numbering
+type BitNumbering int
+
+const (
+	// LSB0 - bit ordering starts with the low-order bit
+	LSB0 BitNumbering = iota
+
+	// MSB0 - bit ordering starts with the high-order bit
+	MSB0
+)
+
+// BitVector is used to manipulate ordered collections of bits
+type BitVector struct {
+	BytePacking BitNumbering
+	Buf         []byte
+	Len         int
```

Comment:
Please document these:
// BytePacking is the bit ordering within bytes
// Len is the logical number of bits in the vector. The last byte in Buf may have undefined bits if Len is not a multiple of 8.